devcontainers/lib/dependabot/devcontainers/file_parser/feature_dependency_parser.rb

```diff
@@ -0,0 +1,105 @@
+# typed: true
+# frozen_string_literal: true
+
+require "dependabot/devcontainers/requirement"
+require "dependabot/file_parsers/base"
+require "dependabot/shared_helpers"
+require "dependabot/dependency"
+require "json"
+require "uri"
+
+module Dependabot
+  module Devcontainers
+    class FileParser < Dependabot::FileParsers::Base
+      class FeatureDependencyParser
+        def initialize(config_dependency_file:, repo_contents_path:, credentials:)
+          @config_dependency_file = config_dependency_file
+          @repo_contents_path = repo_contents_path
+          @credentials = credentials
+        end
+
+        def parse
+          SharedHelpers.in_a_temporary_repo_directory(base_dir, repo_contents_path) do
+            File.write(config_name, config_contents)
+            # TODO: Also write lockfile if it exists?
+            SharedHelpers.with_git_configured(credentials: credentials) do
+              parse_cli_json(evaluate_with_cli)
+            end
+          end
+        end
+
+        private
+
+        def base_dir
+          config_dependency_file.directory
+        end
+
+        def config_name
+          config_dependency_file.name
+        end
+
+        def config_contents
+          config_dependency_file.content
+        end
+
+        # https://github.com/devcontainers/cli/blob/9444540283b236298c28f397dea879e7ec222ca1/src/spec-node/devContainersSpecCLI.ts#L1072
+        def evaluate_with_cli
+          raise "config_name must be a string" unless config_name.is_a?(String) && !config_name.empty?
+
+          cmd = "devcontainer outdated --workspace-folder . --config #{config_name} --output-format json"
+          puts "Running command: #{cmd}"
+          begin
+            json = SharedHelpers.run_shell_command(
+              cmd,
+              stderr_to_stdout: false
+            )
+            JSON.parse(json)
+          rescue StandardError => e
+            puts "Error running shell command:\n#{e.message}"
+            Dependabot.logger.info("Failed to run 'devcontainer outdated' command:\n#{e.message}")
```

Comment:
This is an interesting technique. It allows us to mostly skip the "update checker" step, which seems nice.

Using this approach only dependencies that have updates available get parsed, correct? So an update would never result in "no updates needed" because up to date dependencies are "invisible", correct?